doc: cover more cache modes in rados/operations/cache-tiering.rst

smithfarm opened the pull request:
Fixes: http://tracker.ceph.com/issues/14153
Signed-off-by: Nathan Cutler <ncutler@suse.com>

smithfarm commented:
Tried to make it consistent with http://docs.ceph.com/docs/master/dev/cache-pool/#cache-mode

liewegas commented on doc/rados/operations/cache-tiering.rst:
```diff
@@ -62,6 +62,28 @@ configure how this migration takes place. There are two main scenarios:
+  In other words, read requests that miss in the cache are forwarded to the
+  base tier. Write operations on the cache are accepted and flushed to the base
+  tier. This mode is useful when the base and cache tiers are comparably fast
+  for reads, but there is asymmetric performance or endurance for writes.
```
I think we should not mention the 'forward' modes, and (ideally) we should remove them entirely.  They do not work right.  We can make exsiting cache tiers using it switch to read-proxy on upgrade.

liewegas commented:
IIRC most of these modes are marked as experimental and you ahve to pass --force to enable them?  Maybe they should be flagged that way here?

tchaikov commented:
@smithfarm ping?

smithfarm commented:
@tchaikov Removing the forward modes entirely, and making existing cache tiers switch to read-proxy on upgrade, is beyond the scope of this PR. Care to open a companion PR?

tchaikov commented:
@smithfarm agreed. could you address sage's comment before we merge this PR?

> A more obvious/stronger warning on read-only is warranted, though.. my guess is most users would either not think about it or expect readonly to mean a write evicts the object.

we could highlight this using 
```rst
.. warning::

```

or just highlight the "no" in 

> Ceph makes no attempt to sync these updates to the corresponding objects in the cache.

gregsfortytwo commented:
If these comments are no longer good, we probably want new ones. @smithfarm? :)

liewegas commented:
unstale

smithfarm commented:
Sigh. Sorry this has taken so long.

smithfarm commented:
Changelog:

* added more obvious/stronger warning on read-only
* dropped read-forward and forward - i.e. the documentation will not mention these modes at all, even though they are present in the code

Still to be done:

* determine whether `--force` is required and mention that, if it is

smithfarm commented:
```
target192168000083:/home/ubuntu # ceph osd tier add backing tier
pool 'tier' is now (or already was) a tier of 'backing'
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier none
set cache-mode for pool 'tier' to none
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier writeback
set cache-mode for pool 'tier' to writeback
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier forward
Error EPERM: 'forward' is not a well-supported cache mode and may corrupt your data.  pass --yes-i-really-mean-it to force.
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier readonly
Error EPERM: 'readonly' is not a well-supported cache mode and may corrupt your data.  pass --yes-i-really-mean-it to force.
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier readforward
Error EPERM: 'readforward' is not a well-supported cache mode and may corrupt your data.  pass --yes-i-really-mean-it to force.
target192168000083:/home/ubuntu # ceph osd tier cache-mode tier readproxy
set cache-mode for pool 'tier' to readproxy
```

smithfarm commented:
Changelog:

* mention that `readonly` cache mode is experimental

@liewegas @tchaikov Please re-review?

liewegas commented:
see https://github.com/ceph/ceph/pull/28944
